File: rust_src/9.回文数.rs
/*
 * @lc app=leetcode.cn id=9 lang=rust
 *
 * [9] 回文数
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
		if x < 0
		{
			return false;
		}
		let s = x.to_string();

		return s.chars().rev().collect::<String>() == s;
    }
}

// @lc code=end

File: python_src/9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#
from typing import *

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x = str(x)
        for i in range(len(x) >> 1):
            if x[i] != x[-i - 1]:
                return False
        return True
        
# @lc code=end

s = Solution()
print(s.isPalindrome(121))
print(s.isPalindrome(-121))
print(s.isPalindrome(10))
print(s.isPalindrome(12321))
print(s.isPalindrome(12321))
